### DISCUSSION TITLE
How to implement animation when code completion happens?

### DISCUSSION THREAD

**Discussion by doggy8088** [community/community#146550] [Apps, API and Webhooks]
I saw this that is very cool. 
https://www.instagram.com/share/BAFs_IMi94

I would like to implement similar function for this but only when code completion happens. Is that possible?

**Comment by M-tech-cmd** [comment]
1. Understand the Feature: First, identify what the Instagram feature does. If it’s a visual effect or a notification, you’ll need to replicate that behavior in your code editor.

2. Choose Your Code Editor: Depending on the code editor you’re using (e.g., Visual Studio Code, IntelliJ IDEA), the implementation will vary. Each editor has its own API for extensions and plugins.

3. Use the Editor’s API: Most modern code editors provide APIs to create extensions or plugins. For example:
Visual Studio Code: You can use the VS Code API to listen for code completion events and trigger your custom function.
[IntelliJ IDEA: You can use the IntelliJ Platform SDK to implement custom code completion behavior](https://www.plugin-dev.com/intellij/custom-language/code-completion/)[1](https://www.plugin-dev.com/intellij/custom-language/code-completion/).

4. Implement the Function: Write the code to trigger your desired function when a code completion event occurs. This might involve:
Listening for code completion events.
Executing your custom function when these events are detected.

5. Test and Refine: Make sure to thoroughly test your implementation to ensure it works as expected and doesn’t interfere with the normal operation of the code editor.
